### PR TITLE
fix: respect --statement-typename-introspection and --statement-max-depth when generating types

### PR DESCRIPTION
**Pull request by dpilch**
## Problem

`--statement-typename-introspection` and `--statement-max-depth` are ignored for type generation resulting in types that do not match the generated statements.

**Issue number, if available:** N/A

## Changes

Pass `--statement-typename-introspection` and `--statement-max-depth` to type generation.

**Corresponding docs PR, if applicable:** N/A

## Validation

Add a new unit test to ensure the values are passed correctly.

## Checklist

- [x] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [x] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [x] If this PR requires a docs update, I have linked to that docs PR above.
- [x] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: .changeset/shaggy-olives-melt.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@aws-amplify/model-generator': patch
+---
+
+respect --statement-typename-introspection and --statement-typename-introspection when generating types
```
Comment (edwardfoyle) [inline .changeset/shaggy-olives-melt.md:5]: I'm assuming this should be `--statement-typename-introspection and --max-depth`?

**Review by phani-srikar — APPROVED** [review]

**Review by edwardfoyle — COMMENTED** [review]
So these args were already present, they were just being wired into codegen but not modelgen?

File: packages/model-generator/API.md [hunk 2]
```diff
@@ -196,7 +196,7 @@ export type ModelsGenerationParameters = {
 export type TypesGenerationParameters = {
     target: TypesTarget;
     multipleSwiftFiles?: boolean;
-};
+} & Pick<DocumentGenerationParameters, 'maxDepth' | 'typenameIntrospection'>;
```
Comment (sobolk) [inline packages/model-generator/API.md:199]: Wouldn't it be better to just declare two properties inline?
It takes additional jump for somebody reading this declaration to understand what `TypesGenerationParameters` is if we use `Pick`

Comment (dpilch) [reply packages/model-generator/API.md:199]: I did this to keep the types in sync, but I suppose the TypeScript build will already handle that with inline properties as well.

**Comment by dpilch** [comment]
Not modelgen. There was different behavior for codegen statements and codegen types. In gen 1 the codegen for types would read the generated statements from the file system. Gen 2 is not using the file system so generation for types will first regenerate the statements and store in memory to use an input for type generation. The generated statements in this case were not following the `maxDepth` and `typenameIntrospection` parameters.

https://github.com/aws-amplify/amplify-backend/pull/1392/files#diff-1e88024fbe426c6c75d6cfd357396c835499075b6a1878a2d9858cdc027050eeR36-R52

File: .changeset/shaggy-olives-melt.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@aws-amplify/model-generator': patch
```
Comment (edwardfoyle) [inline .changeset/shaggy-olives-melt.md:2]: nit:
```suggestion
'@aws-amplify/model-generator': minor
```

**Review by edwardfoyle — APPROVED** [review]

**Review by sobolk — APPROVED** [review]

**Review by edwardfoyle — APPROVED** [review]

**Comment by dpilch** [comment]
Seems like there is a bug in `check_pr_size`. It is claiming:
```
Lines Added: 1092
Lines Removed: 377
```

Is there a fix for this?

**Comment by sobolk** [comment]
> Seems like there is a bug in `check_pr_size`. It is claiming:
> 
> ```
> Lines Added: 1092
> Lines Removed: 377
> ```
> 
> Is there a fix for this?

This happens if PR is behind `main`. If you pull latest main it should get green. Conflictless main merge shouldn't reset approvals.

**Pull request merged**